### ISSUE TITLE
Disposing of an observable involving a ReactiveCommand will leave the ReactiveCommand with CanExecute set to false

### ISSUE THREAD

**Issue by PureWeen** [reactiveui/ReactiveUI#1525]
**Do you want to request a *feature* or report a *bug*?**
Bug (possibly expected behavior)

**What is the current behavior?**
If you dispose of the observable returned from
```
ReactiveCommand.Execute()
```

while it's executing, the command won't transition back to IsExecuting False.  Which means future setups with things like InvokeCommand are basically dead in the water.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**

```
   [Fact]
        public void IsExecutingIsFalseAfterCommandDisposal()
        {
            (new TestScheduler()).With(sched => {
                
                int executeCount = 0;
                var execute = Observables.Unit.Delay(TimeSpan.FromSeconds(1), sched);
                var fixture = ReactiveCommand.CreateFromObservable(() => {
                    executeCount++;
                    return execute;
                }, outputScheduler: sched);
                var isExecuting = fixture
                    .IsExecuting
                    .CreateCollection(scheduler: ImmediateScheduler.Instance);
                
                var source = new Subject<Unit>();

                var disp = 
                    source
                        .InvokeCommand(fixture);


                source.OnNext(Unit.Default);
                
                sched.AdvanceByMs(100);
                Assert.Equal(2, isExecuting.Count);
                Assert.False(isExecuting[0]);
                Assert.True(isExecuting[1]);
                Assert.Equal(1, executeCount);
                
                //Dispose of execute
                disp.Dispose();
                sched.AdvanceByMs(901);

                //isexecuting is now stuck set to true until next executed
                //Assert.Equal(3, isExecuting.Count);
               // Assert.False(isExecuting[2]);


                source.OnNext(Unit.Default);
                sched.AdvanceByMs(100);
                Assert.Equal(2, executeCount); //FAILS
                
            });
        }
```

**What is the expected behavior?**

For a command that's disposed of to transition IsExecuting back to false

**What is the motivation / use case for changing the behavior?**

A lot of setups for RxUI take the form

```
this.WhenActivated(disp
...
this.WhenAnyValue(x=> x)
.InvokeCommand(cmd)
.DisposeWith(disp)
```

If the cmd is in the middle of executing and you navigate away from the page then come back to the page then that setup is basically now just dead in the water as the cmd will just sit there with "IsExecuting == true" and "CanExecute == false"

I was able to surface this issue in the Cinephile sample application by adding a 10 second Delay to the ReactiveCommand used in that app

```
  (movies, genres) =>
                    {
                        return movies
                                .Results
                                .Select(movieDto => MapDtoToModel(genres, movieDto));
                    })
                    .Delay(TimeSpan.FromSeconds(10));
```

- I set the pagesize to like 5 so the command would trigger multiple times
- Scrolled down to trigger the command
- clicked on a movie before the command could finish
- Navigated back to main screen
- now when you scroll and it tries to Invoke "InvokeCommand(LoadMovies)" it is no longer able to invoke it

**Which versions of ReactiveUI, and which platform / OS are affected by this issue? Did this work in previous versions of ReativeUI? Please also test with the [latest stable](https://www.nuget.org/profiles/reactiveui) and [development snapshot](https://reactiveui.net/docs/getting-started/installation/nuget-packages/development-package-feed)**

I wrote the included unit test against the develop branch. I saw this behavior in 7.4.0 as well

**Other information (e.g. stacktraces, related issues, suggestions how to fix)**
Not sure how to fix off hand ... Not sure if it's just expected behavior

**Comment by kentcb** [comment]
Just a note to say I've seen this, I want to try and repro and fix, but I can't build the source right now.

**Comment by kentcb** [comment]
OK, I have fixed this as part of #1534. Bad form, I know, but all this stuff was interrelated.
